src/sentry/sentry_metrics/extraction_rules.py

```diff
@@ -0,0 +1,114 @@
+from collections.abc import Mapping, Sequence
+from dataclasses import dataclass
+from typing import Any
+
+from sentry.models.project import Project
+from sentry.utils import json
+
+METRICS_EXTRACTION_RULES_OPTION_KEY = "sentry:metrics_extraction_rules"
+
+
+class MetricsExtractionRuleValidationError(ValueError):
+    pass
+
+
+@dataclass(frozen=True)
+class MetricsExtractionRule:
+    span_attribute: str
+    type: str
+    unit: str
+    tags: set[str]
+
+    @classmethod
+    def from_dict(cls, dictionary: Mapping[str, Any]) -> "MetricsExtractionRule":
+        return MetricsExtractionRule(
+            span_attribute=dictionary["span_attribute"],
```

Comment:
nit: afaik we try to use camel case in the request/response objects
```suggestion
            span_attribute=dictionary["spanAttribute"],
```